homeassistant/components/skybell/__init__.py

```diff
@@ -1,101 +1,162 @@
+            for coordinator in device_coordinators
+        ]
+    )
+    hass.data[DOMAIN][entry.entry_id] = device_coordinators
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
     return True
 
 
-class SkybellDevice(Entity):
-    """A HA implementation for Skybell devices."""
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
+    return unload_ok
+
 
-    def __init__(self, device):
-        """Initialize a sensor for Skybell device."""
-        self._device = device
+class SkybellEntity(CoordinatorEntity[SkybellDataUpdateCoordinator]):
+    """An HA implementation for Skybell entity."""
 
-    def update(self):
-        """Update automation state."""
-        self._device.refresh()
+    _attr_attribution = "Data provided by Skybell.com"
+
+    def __init__(self, coordinator: SkybellDataUpdateCoordinator) -> None:
+        """Initialize a SkyBell entity."""
+        super().__init__(coordinator)
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.coordinator.device.device_id)},
+            manufacturer=DEFAULT_NAME,
+            model=self.coordinator.device.type,
+            name=self.coordinator.device.name,
+            sw_version=self.coordinator.device.firmware_ver,
+        )
+        if self.coordinator.device.mac:
+            self._attr_device_info[ATTR_CONNECTIONS] = {
+                (dr.CONNECTION_NETWORK_MAC, self.coordinator.device.mac)
+            }
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, str | int | tuple[str, str]]:
         """Return the state attributes."""
-        return {
-            ATTR_ATTRIBUTION: ATTRIBUTION,
-            "device_id": self._device.device_id,
-            "status": self._device.status,
-            "location": self._device.location,
-            "wifi_ssid": self._device.wifi_ssid,
-            "wifi_status": self._device.wifi_status,
-            "last_check_in": self._device.last_check_in,
-            "motion_threshold": self._device.motion_threshold,
-            "video_profile": self._device.video_profile,
+        attr: dict[str, str | int | tuple[str, str]] = {
+            "device_id": self.coordinator.device.device_id,
+            "status": self.coordinator.device.status,
+            "location": self.coordinator.device.location,
+            "motion_threshold": self.coordinator.device.motion_threshold,
+            "video_profile": self.coordinator.device.video_profile,
         }
+        if self.coordinator.device.owner:
+            attr["wifi_ssid"] = self.coordinator.device.wifi_ssid
+            attr["wifi_status"] = self.coordinator.device.wifi_status
+            attr["last_check_in"] = self.coordinator.device.last_check_in
```

Comment:
For a future PR

Entities that generate a significant amount of state changes can quickly increase the size of the database when the extra_state_attributes also change frequently. Minimize the number of extra_state_attributes for these entities by removing non-critical attributes or creating additional sensor entities.

https://developers.home-assistant.io/docs/core/entity/?_highlight=extra_state_attributes